sentinelhub/config.py

```diff
@@ -11,7 +11,116 @@
 ConfigDict = Dict[str, Union[str, int, float]]
 
 
-class SHConfig:
+class _BaseSHConfig:
+    """A base class for the sentinelhub-py package configuration class.
+
+    Only contains defaults and custom getter methods.
+    """
+
+    CREDENTIALS = {
+        "instance_id",
+        "sh_client_id",
+        "sh_client_secret",
+        "aws_access_key_id",
+        "aws_secret_access_key",
+        "aws_session_token",
+    }
+    OTHER_PARAMS = {
+        "sh_base_url",
+        "sh_auth_base_url",
+        "geopedia_wms_url",
+        "geopedia_rest_url",
+        "aws_metadata_url",
+        "aws_s3_l1c_bucket",
+        "aws_s3_l2a_bucket",
+        "opensearch_url",
+        "max_wfs_records_per_query",
+        "max_opensearch_records_per_query",
+        "max_download_attempts",
+        "download_sleep_time",
+        "download_timeout_seconds",
+        "number_of_download_processes",
+    }
```

Comment:
I figured that might be an issue. Technically both can be lists and then `CONFIG_PARAMS` merges them together in the correct order, i.e. `OTHER_PARAMS[:3]+CREDENTIALS+OTHER_PARAMS[3:]` to avoid duplication.

Reply:
I think the order is a bit more mixed. Because it is not very explicit I have a feeling it could eventually cause confusion and problems. :thinking: